File: Producto/backend/src/main/java/backend/com/produccion/application/dto/DespachoOSDTO.java
package backend.com.produccion.application.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class DespachoOSDTO {

    private Long idDespacho;
    private Long osId;
    private LocalDate fechaDespacho;
    private Integer cantidadDespachada;
    private String descripcion;
    private String responsable;
    private String observaciones;
}

File: Producto/frontend/src/remote/DTO/DespachoOSDTO.js
/**
 * DTO para representar un despacho de OS (envío al taller externo).
 * Espejo de DespachoOSDTO.java del backend.
 */
export class DespachoOSDTO {
    constructor(data = {}) {
        this.idDespacho = data.idDespacho ?? null;
        this.osId = data.osId ?? null;
        this.fechaDespacho = data.fechaDespacho ?? null;
        this.cantidadDespachada = data.cantidadDespachada ?? 0;
        this.descripcion = data.descripcion ?? '';
        this.responsable = data.responsable ?? '';
        this.observaciones = data.observaciones ?? '';
    }
}
